core/src/main/java/cz/o2/proxima/scheme/AttributeValueAccessors.java

```diff
@@ -0,0 +1,297 @@
+     *
+     * @param value input value
+     * @return value as string
+     */
+    String valueOf(T value);
+
+    /**
+     * Return enum value from string
+     *
+     * @param value string representation of enum value.
+     * @return enum value
+     */
+    T createFrom(String value);
+
+    @Override
+    default ValueAccessorType getType() {
+      return ValueAccessorType.ENUM;
+    }
+  }
+
+  /**
+   * Default implementation of {@link ArrayValueAccessor}.
+   *
+   * <p>This implementation delegate {@link #createFrom(Object[])} and {@link #valuesOf(Object[])}
+   * to value accessor.
+   *
+   * @param <T> array value type
+   */
+  public static class DefaultArrayValueAccessor<T> implements ArrayValueAccessor<T> {
+
+    private final DelegateToAccessor<T> delegateToAccessor;
+
+    public DefaultArrayValueAccessor(ValueAccessor valueAccessor) {
+      delegateToAccessor = createDelegates(valueAccessor);
+    }
+
+    private DelegateToAccessor<T> createDelegates(ValueAccessor accessor) {
+      switch (accessor.getType()) {
+        case PRIMITIVE:
+          return new DelegateToAccessor<T>() {
+            @SuppressWarnings("unchecked")
+            final PrimitiveValueAccessor<T> cast = (PrimitiveValueAccessor<T>) accessor;
+
+            @Override
+            public T createFrom(Object object) {
+              return cast.createFrom(object);
+            }
+
+            @Override
+            public <V> V valueOf(T object) {
+              return cast.valueOf(object);
+            }
+          };
+        case STRUCTURE:
+          return new DelegateToAccessor<T>() {
+            @SuppressWarnings("unchecked")
+            final StructureValueAccessor<T> cast = (StructureValueAccessor<T>) accessor;
+
+            @Override
+            @SuppressWarnings("unchecked")
+            public T createFrom(Object object) {
+              return cast.createFrom((Map<String, Object>) object);
+            }
+
+            @Override
+            @SuppressWarnings("unchecked")
+            public <V> V valueOf(T object) {
+              return (V) cast.valuesOf(object);
+            }
+          };
+        case ENUM:
+          return new DelegateToAccessor<T>() {
+            @SuppressWarnings("unchecked")
+            final EnumValueAccessor<T> cast = (EnumValueAccessor<T>) accessor;
+
+            @Override
+            public T createFrom(Object object) {
+              return cast.createFrom(object.toString());
+            }
+
+            @Override
+            @SuppressWarnings("unchecked")
+            public <V> V valueOf(T object) {
+              return (V) cast.valueOf(object);
+            }
+          };
+        default:
+          throw new UnsupportedOperationException("Unknown accessor type " + accessor.getType());
+      }
+    }
+
+    /**
+     * Delegation interface
+     *
+     * @param <T>
+     */
+    private interface DelegateToAccessor<T> extends Serializable {
```

Comment:
Looks like this could be unified with the modified `ValueAccessor`. At least its unclear to me what the difference is.